### PR TITLE
HIVE-25692: ExceptionHandler may mask checked exceptions

### PR DESCRIPTION
**Pull request by kgyrtkirk**

### CODE REVIEW

**Comment by kgyrtkirk** [comment]
current version also contains HIVE-25634

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java [hunk 1]
```diff
@@ -55,13 +54,48 @@ public static ExceptionHandler handleException(Exception e) {
...
+  }
+
+  /**
+   * Throws if the input exception is the instance of the one in the input classes
+   */
+  public <T1 extends Exception,
+          T2 extends Exception,
+          T3 extends Exception,
+          T4 extends Exception>
+  ExceptionHandler throwIfInstance(
```
Comment (dengzhhu653) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java:90]: Nit: maybe we do not need this method with four arguments,  the other three methods seem to be enough for handling exceptions.

**Review by dengzhhu653 — APPROVED** [review]

**Comment by dengzhhu653** [comment]
@kgyrtkirk Thank you for the patch, changes look good to me!

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java (hunk 1 above)
Comment (kgyrtkirk) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java]: you are right - when I suspected this issue I changed one of the call sites which already had 3 arguments ...but then undone that change...

removed

**Pull request merged**